### PR TITLE
Ubloxd: glonass parsing tests

### PR DESCRIPTION
**Pull request by gast04**
contains final glonass kaita version including tests

### CODE REVIEW

File: selfdrive/locationd/test/test_runner.cc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#define CATCH_CONFIG_MAIN
+#include "catch2/catch.hpp"
```
Comment (adeebshihadeh) [inline selfdrive/locationd/test/test_runner.cc:2]: give this file a better name. there's lot of different things in `selfdrive/locationd`

File: selfdrive/locationd/test/test_glonass_kaitai.cc [hunk 2]
```diff
@@ -0,0 +1,368 @@
+#include <iostream>
+#include <vector>
+#include <bitset>
+#include <cassert>
+#include <cstdlib>
+#include <ctime>
+
+#include "catch2/catch.hpp"
+#include "../generated/glonass.h"
```
Comment (adeebshihadeh) [inline selfdrive/locationd/test/test_glonass_kaitai.cc:9]: paths should always be relative to the openpilot root
```suggestion
#include "selfdrive/locationd/generated/glonass.h"
```

File: selfdrive/locationd/test/test_glonass_kaitai.cc [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+using namespace std;
```
Comment (adeebshihadeh) [inline selfdrive/locationd/test/test_glonass_kaitai.cc:11]: we avoid this openpilot

File: selfdrive/locationd/test/test_glonass_kaitai.cc [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+    data.push_back({2, 1}); // p1
+    data.push_back({12, 113}); // t_k
+    data.push_back({1, rand() & 1}); // x_vel_sign
+    data.push_back({23, 7122}); // x_vel_value
+    data.push_back({1, rand() & 1}); // x_accel_sign
+    data.push_back({4, 3}); // x_accel_value
+    data.push_back({1, rand() & 1}); // x_sign
+    data.push_back({26, 33554431}); // x_value
+  }
+  else if (string_number == 2) {
```
Comment (adeebshihadeh) [inline selfdrive/locationd/test/test_glonass_kaitai.cc:130]: else goes on the same line as the bracket

File: selfdrive/locationd/test/test_glonass_kaitai.cc [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+  REQUIRE(gl_string.hamming_code() == data[ST6_HC_OFF + HC_IDX].second);
+  REQUIRE(gl_string.pad_1() == data[ST6_HC_OFF + PAD1_IDX].second);
+  REQUIRE(gl_string.superframe_number() == data[ST6_HC_OFF + SUPERFRAME_IDX].second);
+  REQUIRE(gl_string.pad_2() == data[ST6_HC_OFF + PAD2_IDX].second);
+  REQUIRE(gl_string.frame_number() == data[ST6_HC_OFF + FRAME_IDX].second);
+
+  kaitai::kstream strni(inp_data);
+  glonass_t strni_data(&strni);
+  glonass_t::string_non_immediate_t* sni =
+    static_cast<glonass_t::string_non_immediate_t*>(strni_data.data());
```
Comment (adeebshihadeh) [inline selfdrive/locationd/test/test_glonass_kaitai.cc:364]: why are these on different lines? we don't enforce an 80 char line max

File: selfdrive/locationd/SConscript [hunk 6]
```diff
@@ -13,7 +13,11 @@ if GetOption('kaitai'):
   patch = env.Command(None, 'glonass_fix.patch', 'git apply $SOURCES')
   env.Depends(patch, glonass)
 
-env.Program("ubloxd", ["ubloxd.cc", "ublox_msg.cc", "generated/ubx.cpp", "generated/gps.cpp"], LIBS=loc_libs)
+glonass_obj = env.Object('generated/glonass.cpp')
+if GetOption('test'):
+  env.Program("test/test_runner", ['test/test_runner.cc', 'test/test_glonass_kaitai.cc', glonass_obj], LIBS=[loc_libs])
```
Comment (adeebshihadeh) [inline selfdrive/locationd/SConscript:18]: we generally put the tests at the bottom of the SConscript

File: selfdrive/locationd/test/test_glonass_kaitai.cc (hunk 5 above)
Comment (gast04) [reply selfdrive/locationd/test/test_glonass_kaitai.cc:364]: its just cause I prefer 80chars len

**Pull request merged**